core/camera/src/main/java/com/google/jetpackcamera/core/camera/effects/LowLightBoostSessionContainer.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.core.camera.effects
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.google.android.gms.cameralowlight.LowLightBoostSession
+
+@RequiresApi(Build.VERSION_CODES.R)
+class LowLightBoostSessionContainer {
+    var lowLightBoostSession: LowLightBoostSession? = null
+
+    fun releaseSession() {
+        lowLightBoostSession?.release()
+        lowLightBoostSession = null
+    }
+
+    companion object {
+        private var instance: LowLightBoostSessionContainer? = null
+
+        fun getInstance(): LowLightBoostSessionContainer = instance ?: synchronized(this) {
```

Comment:
I would prefer if we can stay away from using a singleton here. I feel like we should be able to manage LowLightBoostSession lifecycles with the coroutine scopes that they are created in. Maybe that could be addressed in a follow-up PR.

Reply:
Acknowledged. I'll look into this while this PR is under review if there's time. Otherwise I'll address this in a follow-up PR.

Reply:
This was addressed some time ago. Closing this comment now.